app/domain/synchronize/mailchimp/subscriber_tagger.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+#
+#  Copyright (c) 2012-2020, CVP Schweiz. This file is part of
+#  hitobito_cvp and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_cvp.
+
+module Synchronize
+  module Mailchimp
+    class SubscriberTagger
+      attr_reader :emails, :list
+
+      def initialize(emails, list)
+        @emails = emails
+        @list = list
+      end
+
+      def tag!
+        emails.each do |email|
+          subscriber = subscribers[email]
+          next unless subscriber
+
+          kind = subscriber.primary? ? :primary : :additional
+          Contactable::InvalidEmailTagger.new(subscriber.person, email, kind).tag!
+        end
+      end
+
+      def subscribers
+        @subscribers ||= Synchronize::Mailchimp::Subscriber
+          .mailing_list_subscribers(list).index_by(&:email)
```

Comment:
Vielleicht verstehe ich falsch, was das `cleaned` flag bedeutet. Aber wäre es nicht denkbar dass mehrere Personen in der Liste die gleiche `additional_email` haben, aber nur eine Person davon in Mailchimp als `cleaned` auftaucht? Wenn ja, sollten wir vielleicht nicht die betroffenen cleaned `members` in `emails` umrechnen und danach wieder zurück in `mailing_list_subscribers`.

Reply:
Vorerst denke ich können wir das ausklammern da status `cleaned` bedeutet, dass die Email hard bounces hat, also als ungültig betrachtet werden kann. 

Ich denke es ist aktuell garnicht möglich (bzw nicht defniert) was passiert, wenn in einer Liste eine Email Adresse mehrfach verwendet wird, da wir [die subscriber id aus der email ableiten]( https://github.com/hitobito/hitobito/blob/1422c6dcc8fa4034b31abeb96efc3aee5159ea6c/app/domain/synchronize/mailchimp/client.rb#L157). Die mehrfach emails könnten wir auch aus einer Kombination von `primary` und `additional` bekommen.